pinot-spi/src/main/java/org/apache/pinot/spi/config/table/IndexingConfig.java

```diff
@@ -55,10 +55,11 @@ public class IndexingConfig extends BaseJsonConfig {
   private boolean _nullHandlingEnabled;
 
   /**
-   * If `optimizeDictionaryForMetrics` enabled, dictionary is not created for the metric columns
-   * for which rawIndexSize / forwardIndexSize is less than the `noDictionarySizeRatioThreshold`.
+   * If `optimizeDictionaryForMetrics` enabled, dictionary is not created for the high-cardinality
+   * single-valued columns for which rawIndexSize / forwardIndexSize is less than the `noDictionarySizeRatioThreshold`.
+   * It also disables dictionary for json and text columns.
    */
-  private boolean _optimizeDictionaryForMetrics;
+  private boolean _optimizeDictionary;
```

Comment:
Why do we need config for that? Isn't it determined by data type of the column?

Reply:
We want a config to give more control to the user so that they can explicitly choose the index to create if desired. In certain scenarios, raw index might be able to save space, but not good for query performance

Reply:
I think that is out of scope for this PR. That will anyways not be a boolean config so we can decide on that later. The config name for that would be more like `dictionaryType`